code/modules/power/singularity/particle_accelerator/particle.dm

```diff
@@ -65,6 +65,10 @@
 	if(QDELETED(src))
 		return
 	if(!step(src,dir))
+		var/next_step = get_step(src, dir)
+		if(!next_step) // reached the edge of the map
+			qdel(src)
+			return
```

Comment:
Then use next_step in the forcemove proc?

Reply:
forgor